Formula/meson.rb

```diff
@@ -0,0 +1,29 @@
+class Meson < Formula
+  desc "Fast and user friendly build system"
+  homepage "http://mesonbuild.com/"
+  url "https://github.com/mesonbuild/meson/releases/download/0.31.0/meson-0.31.0.tar.gz"
+  sha256 "803d7aa99329c5439ef971c1dcaa39de12d9208f82f0ed029e3e8f9c13ef93f0"
+
+  depends_on :python3
+  depends_on "ninja"
+
+  def install
+    version = Language::Python.major_minor_version("python3")
+    ENV["PYTHONPATH"] = lib+"python#{version}/site-packages"
+
+    system "python3", *Language::Python.setup_install_args(prefix)
+
+    bin.env_script_all_files(libexec+"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
+
+  test do
+    (testpath/"helloworld.c").write("main() { puts(\"hi\"); return 0; }")
```

Comment:
Style nit more than anything but these will be a bit easier to read if you use `<<-EOS.undent`. If you `grep` or `ack` or whatever your favourite search tool is for usage you'll find some solid examples.